### ISSUE TITLE
Alignment of positive spikes

### ISSUE THREAD

**Issue by shihaisun** [cortex-lab/KiloSort#181]
Hi,

I was wondering how positive spikes are aligned with spike time? To my understanding, the spike time of negative spikes indicates the spike's minimum trough. However, this doesn't seem to be the case for positive spikes. In the image below, I extracted the waveform of a single cluster from the raw binary file. The three above are individual spikes and the one below is an average of 10,000 random spikes from that cluster. The x-axes represent time in sample with the 31st sample representing the spike time (red line). So it seems that the positive peak of the spikes are aligned but not at the spike time?

![positivewaveforms](https://user-images.githubusercontent.com/30403651/53796518-74d92200-3f88-11e9-87c2-798da49e6939.png)

Spike times were extracted from spike_times.npy and waveforms from the binary file. I used the getWaveforms.m script by C. Schoonover and A. Fink to extract the waveforms.

So in short, just wanted to know how positive spikes are aligned with the spike time.

Thank you for your time,
Scott

**Comment by adehad** [comment]
I don't believe you can get positive spikes to align to the peak in KiloSort, as the code looks for a local minimum around the threshold crossing:
https://github.com/cortex-lab/KiloSort/blob/3ff2d1027166b6e54ab4661b8c56e33e1ba9c4e7/preProcess/isolated_peaks.m#L4-L5

In our usage case, we multiplied the data by -1 and ran this new inverted data through KiloSort to get the correct spike times. Alternatively, you can try looking for a maximum within a certain number of samples from KiloSort's spiketimes (e.g. like how KiloSort uses `loc_range`) .
